### PR TITLE
Add new scss stylesheet for GithubCorner

### LINKED ISSUE #130: Update Github-Corner stylesheets
**Description**
Update the stylesheets for the `GithubCorner` component defined in [vue-vocabulary](https://github.com/creativecommons/vue-vocabulary) in accordance with changes requested in [#118](https://github.com/creativecommons/vue-vocabulary/issues/118) and discussed with @dhruvkb 

**Problems solved**
Updates the configuration of the GithubCorner component as suggested by @annatuma 

**Alternatives**

**Additional context**
Changes were requested in the component that required corresponding alterations to the stylesheets defined in Vocabulary itself. The aim is to update the styles so that `GithubCorners` has only 2 color variants.

**Want to take up?**
Yes

### PR DESCRIPTION
**Pull request by jshreyans**
**Fixes**
Fixes #130  by @jshreyans

**Description**
This PR adds a new SCSS file to the `/styles` directory specifically for the GithubCorner component from vue-vocabulary. For further context, refer to [#118](https://github.com/creativecommons/vue-vocabulary/issues/118) and [#155](https://github.com/creativecommons/vue-vocabulary/pull/155). After changes from this issue, the styles needed to be modified so that the GithubCorner component only had 2 colors and 2 positions.

**Type of PR** 
This PR is a refactor.

**Technicalities**
For styling reference, the `.styl` file that was made long back for this component (as part of vue-vocabulary) has been used. The relevant commit for that file can be found [here](https://github.com/creativecommons/vue-vocabulary/commit/5d878281204cdc41e7dc54fb2c0ad5ffc818d929#diff-fab1eda801d1c7d086198302a02433ae)

**Tests**

**Screenshots**

**Checklist:**
 
- [x] My pull request has a descriptive title (not a vague title like "Update `index.md`").
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

**Legal**
By submitting this PR, I agree to abide by the terms of the Developer 
Certificate of Origin.

<details>
<summary>Developer Certificate of Origin</summary>

Developer Certificate of Origin<br>
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.<br>
1 Letterman Drive<br>
Suite D4700<br>
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
@jshreyans the tests are failing. Could you pull `master` into the branch, or alternatively rebase this branch onto `master`? Passing the tests would give some us confidence to merge.

File: src/styles/github_corner.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+@import "./colors.scss";
+
+.vocab.github-corner {
+  --github-corner-primary: $color-black;
+  --github-corner-secondary: $color-white;
+
+  &.inverted {
```
Comment (dhruvkb) [inline src/styles/github_corner.scss]: Modifiers start with `is-` or `has-`. So a better class name would be `is-inverted`.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Add the story to demonstrate. You can paste the SVG from Vue Vocabulary directly into the story.

**Comment by jshreyans** [comment]
> @jshreyans the tests are failing. Could you pull `master` into the branch, or alternatively rebase this branch onto `master`? Passing the tests would give some us confidence to merge.

I pulled master and the tests seem to have passed

File: src/styles/github_corner.scss (hunk 1 above)
Comment (jshreyans) [reply src/styles/github_corner.scss]: Corrected

**Comment by jshreyans** [comment]
> Add the story to demonstrate. You can paste the SVG from Vue Vocabulary directly into the story.

Need a little help with this. I created a story but I seem to have made some mistake. Could you just outline what all the story needs to include?

**Comment by dhruvkb** [comment]
@jshreyans the story should demonstrate
- both black-on-white and white-on-black versions
- both left-aligned and right-aligned versions

**Comment by jshreyans** [comment]
@dhruvkb I've added the story with the above components. Although, I'm a little doubtful whether the left corner position works fine. Because when the corner is on the left, it needs to be spun around along its axis.

File: src/stories/github_corner.stories.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      fill="currentColor"
+      style="transform-origin: 130px 106px;"
+      class="octo-arm">
+    </path>
+    <path
+      d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
+      fill="currentColor"
+      class="octo-body">
+    </path>
+</svg>
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]: Make this SVG an asset in the dir `src/assets/`. Would make it easier to use in Vocabulary as well as Vue Vocabulary.

File: src/stories/github_corner.stories.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+`
+
+export const GithubCorner = () => `
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]:
```suggestion
export const Default = () => `
```

File: src/stories/github_corner.stories.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  :class="gitHubCornerClasses"
+  :href="url"
+  :target="target"
+  :rel="rel"
+  aria-label="View source on GitHub">
+    ${GithubCornerSVG}
+</a>
+`
+
+export const GithubCornerInverted = () => `
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]:
```suggestion
export const Inverted = () => `
```

File: src/stories/github_corner.stories.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  :class="gitHubCornerClasses"
+  :href="url"
+  :target="target"
+  :rel="rel"
+  aria-label="View source on GitHub">
+    ${GithubCornerSVG}
+</a>
+`
+
+export const GithubCornerLeft = () => `
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]:
```suggestion
export const LeftAligned = () => `
```

File: src/stories/github_corner.stories.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  :class="gitHubCornerClasses"
+  :href="url"
+  :target="target"
+  :rel="rel"
+  aria-label="View source on GitHub">
+    ${GithubCornerSVG}
+</a>
+`
+
+export const GithubCornerRight = () => `
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]: Isn't this already demonstrated by the default story?

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: src/stories/github_corner.stories.js (hunk 2 above)
Comment (jshreyans) [reply src/stories/github_corner.stories.js]: Done

File: src/stories/github_corner.stories.js (hunk 6 above)
Comment (jshreyans) [reply src/stories/github_corner.stories.js]: Yeah you're right. Should I remove this variation then?

Comment (dhruvkb) [reply src/stories/github_corner.stories.js]: Yes, remove it. Only keep the distinct variants.

File: src/stories/github_corner.stories.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+</a>
+`
+
+export const Inverted = () => `
+<a
+  class="vocab github-corner is-inverted"
+  :class="gitHubCornerClasses"
+  :href="url"
+  :target="target"
+  :rel="rel"
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]: These attributes that begin with `:` are Vue props, they won't have any effect here. Remove all of them.

File: src/stories/github_corner.stories.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  aria-label="View source on GitHub">
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js:28]: Can also remove this for brevity in examples. Upto you though.

File: src/stories/github_corner.stories.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
+export default { title: 'GithubCorner' }
+
+const GithubCornerSVG = '' +
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]:
```suggestion
const GitHubCornerSVG = '' +
```

File: src/stories/github_corner.stories.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
+export default { title: 'GithubCorner' }
```
Comment (dhruvkb) [inline src/stories/github_corner.stories.js]:
```suggestion
export default { title: 'GitHubCorner' }
```

File: src/styles/github_corner.scss [hunk 11]
```diff
@@ -0,0 +1,44 @@
+@import "./colors.scss";
```
Comment (dhruvkb) [inline src/styles/github_corner.scss]: Not required.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Finish by importing this SCSS file into `src/scss/vocabulary.scss`.

File: src/styles/github_corner.scss (hunk 11 above)
Comment (jshreyans) [reply src/styles/github_corner.scss]: The white and black colors are defined in this file. I thought I should use them for the sake of avoiding repetition. I'll remove it if needed.

File: src/stories/github_corner.stories.js (hunk 8 above)
Comment (jshreyans) [reply src/stories/github_corner.stories.js:28]: I think we should keep it there.

Comment (dhruvkb) [reply src/stories/github_corner.stories.js:28]: Cool, we'll keep it then 👍.

File: src/styles/github_corner.scss (hunk 11 above)
Comment (dhruvkb) [reply src/styles/github_corner.scss]: @jshreyans it's automatically imported when the SCSS is compiled.

Comment (dhruvkb) [reply src/styles/github_corner.scss]: Remove it.

**Review by dhruvkb — COMMENTED** [review]
The import for `colors.scss` is not required as the SCSS when compiled from `vocabulary.scss` automatically resolves these imports.

**Comment by jshreyans** [comment]
Hadn't seen your comment to remove the right aligned variant. This commit should do it.

**Review by dhruvkb — APPROVED** [review]
This is good to go!

**Pull request merged**